src/main/java/com/cbms/source/local/AssetDAOImpl.java

```diff
@@ -0,0 +1,300 @@
+    private static final String GET_ASSET_INFO_FROM_ASSET_ID = "SELECT * FROM attribute_measurements am, attribute att WHERE att.attribute_id=am.attribute_id AND am.asset_id = ?";
+    private static final String GET_ATTRIBUTES_NAMES_FROM_ASSET_ID="SELECT DISTINCT att.attribute_name FROM attribute att, attribute_measurements am, asset a WHERE a.asset_id= ? AND am.asset_id = a.asset_id AND att.attribute_id = am.attribute_id order by att.attribute_id";
+    private static final String GET_ASSETS_FROM_ASSET_TYPE_ID = "SELECT a.asset_id, a.asset_type_id, a.sn, a.location,a.description FROM asset a WHERE a.archived = true AND a.asset_type_id = ?";
+    private static final String GET_ASSET_TYPE_NAME_FROM_ASSET_ID="SELECT at.name FROM asset_type at WHERE at.asset_type_id = ?";
+    private static final String GET_ALL_LIVE_ASSETS="SELECT * FROM asset WHERE archived = false";
+    private static final String INSERT_NEW_ASSET_MEASUREMENT="INSERT INTO asset_model_calculation values( ? , ? ,now(), ?)";
+    private static final String SET_UPDATED_FALSE="UPDATE asset set updated = 0 where asset_id = ?";
+
+    /**
+     * This will return an arraylist of assets that have the updated tag set to true
+     * it will be used to identify which asset need new RUL measurements
+     *
+     * @author Paul
+     */
+    @Override
+    public ArrayList<Asset> getAssetsToUpdate() {
+        ArrayList<Asset> assets = new ArrayList<>();
+        ResultSet rs = nonParamQuery(GET_ASSETS_TO_UPDATE);
+
+        try{
+            while (rs.next()) {
+                assets.add(createAssetFromQueryResult(rs));
+            }
+        }
+        catch (SQLException e){
+            e.printStackTrace();
+        }
+        return assets;
+    }
+
+    /**
+     * When given an asset ID this will delete the the asset from the database as well as the corresponding
+     * tables that reference the asset ID
+     *
+     * @param assetID represents the asset's ID
+     * @author Jeff
+     */
+    @Override
+    public void deleteAssetByID(int assetID) {
+        try {
+            PreparedStatement ps = getConnection().prepareStatement(DELETE_ASSET);
+            ps.setString(1 ," attribute_measurements ");
+            ps.setInt(2 , assetID);
+            ps.executeQuery();
+        }
+        catch (SQLException e) {
+            e.printStackTrace();
+        }
+        try {
+            PreparedStatement ps = getConnection().prepareStatement(DELETE_ASSET);
+            ps.setString(1 ," asset_model_calculation ");
+            ps.setInt(2 , assetID);
+            ps.executeQuery();
+        }
+        catch (SQLException e) {
+            e.printStackTrace();
+        }
+        try {
+            PreparedStatement ps = getConnection().prepareStatement(DELETE_ASSET);
+            ps.setString(1 ," asset ");
+            ps.setInt(2 , assetID);
+            ps.executeQuery();
+        }
+        catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
```

Comment:
Can this be combined into one PreparedStatement instead of querying the database 3 times?

Reply:
ya good point, i'll look into it